src/nvim/msgpack_rpc/channel.c

```diff
@@ -321,13 +332,16 @@ static void parse_msgpack(Channel *channel)
       }
       arena_mem_free(arena_finish(&p->arena));
     } else if (p->type == kMessageTypeResponse) {
-      ChannelCallFrame *frame = kv_last(channel->rpc.call_stack);
-      if (p->request_id != frame->request_id) {
+      const char *client_type = rpc_client_type(channel);
+      ChannelCallFrame *frame = strcmp(client_type, "msgpack-rpc") == 0
```

Comment:
seems weird to do string parsing at _this_ point in the codebase. I think we should use an enum, and parse the string right after the client calls `nvim_set_client_info()` .